sts/sts-gcp/src/main/java/com/salesforce/multicloudj/sts/gcp/GcpSts.java

```diff
@@ -43,22 +45,34 @@ public GcpSts() {
     protected StsCredentials getSTSCredentialsWithAssumeRole(AssumedRoleRequest request){
         GenerateAccessTokenRequest accessTokenRequest = GenerateAccessTokenRequest.newBuilder()
                 .setName("projects/-/serviceAccounts/" + request.getRole())
-                .addAllScope(List.of("https://www.googleapis.com/auth/iam"))
+                .addAllScope(List.of(scope))
                 .setLifetime(Duration.newBuilder().setSeconds(request.getExpiration()).build())
                 .build();
 
         GenerateAccessTokenResponse response = this.stsClient.generateAccessToken(accessTokenRequest);
-        return new StsCredentials(null, null, (response.getAccessToken()));
+        return new StsCredentials(null, null, response.getAccessToken());
     }
 
     @Override
     protected CallerIdentity getCallerIdentityFromProvider() {
-        return null;
+        try {
+            GoogleCredentials credentials = GoogleCredentials.getApplicationDefault().createScoped(List.of(scope));
+            credentials.refreshIfExpired();
+            return new CallerIdentity(null, credentials.getAccessToken().getTokenValue(), null);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
we cannot throw generic runtimeexceptions from provider implementations. We should try to be more explicit and throw one of the best match defined substrate sdk exceptions.